### PR TITLE
HOTFIX: fix NPE in Kafka Streams IQ

### PR DESCRIPTION
**Pull request by mjsax**
Reported on the mailing list. Would be good to get this into 2.5.0, too. If not, also ok. (\cc @arvindth)

Call for review @vvcephei @guozhangwang @brary @vitojeng @vinothchandar @vpapavas

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProviderTest.java [hunk 1]
```diff
@@ -290,6 +292,48 @@ public void shouldReturnEmptyListIfNoStoresFoundWithName() {
             provider.stores(StoreQueryParameters.fromNameAndType("not-a-store", QueryableStoreTypes.keyValueStore())));
     }
 
+    @Test
+    public void shouldReturnSingleStoreForPartition() {
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProviderTest.java:296]: The first two tests are just new test that we missed to add in the original PR.

**Review by vitojeng — APPROVED** [review]
LGTM !

**Comment by guozhangwang** [comment]
LGTM! Just clarifying that this NPE can only be thrown if the user passed in partition number in param that does not exist, right?

**Comment by mjsax** [comment]
> this NPE can only be thrown if the user passed in partition number in param that does not exist, right?

Correct.

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.5`.

**Pull request merged**